src/main/java/com/shipmentEvents/handlers/EventHandler.java

```diff
@@ -64,7 +64,7 @@ public String handleRequest(ScheduledEvent scheduledEvent, Context context) {
     }
 
     public String weakMessageEncryption(String message, String key) throws Exception {
-        Cipher cipher = Cipher.getInstance("RSA");
+        Cipher cipher = Cipher.getInstance("GCM");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It looks like your code uses a cipher object with an insecure transformation. To make your code more secure, use one of the following algorithms with a built-in integrity check: `AES/GCM/NoPadding` or `ChaCha20-Poly1305`.

[Learn more](https://cwe.mitre.org/data/definitions/327.html)

Reply:
Yes! Recommendation generated by Amazon CodeGuru Reviewer!